avm/ptn/sa/conversation-knowledge-mining/main.bicep

```diff
@@ -0,0 +1,1168 @@
+@description('Optional. The name of the docker image for the Rag function.')
+param functionRagDockerImageName string = 'km-rag-function'
+@description('Optional. The tag of the docker image for the Rag function.')
+param functionRagDockerImageTag string = 'latest'
+@description('Optional. The required CPU in cores of the Rag function.')
+param functionRagCpu int = 1
+@description('Optional. The required memory in GiB of the Rag function.')
+param functionRagMemory string = '2Gi'
+@description('Optional. The maximum number of workers that the Rag function can scale out.')
+param functionRagAppScaleLimit int = 10
+@description('Optional. The name of the function to be used to stream text in the Rag function.')
+param functionRagFunctionName string = 'stream_openai_text'
+
+// PARAMETERS: Web app configuration
+@description('Optional. The SKU for the web app. If empty it will be set to B2.')
+param webAppServerFarmSku string = 'B2'
+@description('Optional. The url of the Container Registry where the docker image for Conversation Knowledge Mining webapp is located.')
+param webAppDockerImageContainerRegistryUrl string = 'kmcontainerreg.azurecr.io'
+@description('Optional. The name of the docker image for the Rag function.')
+param webAppDockerImageName string = 'km-app'
+@description('Optional. The tag of the docker image for the Rag function.')
+param webAppDockerImageTag string = 'latest'
+
+// PARAMETERS: Telemetry
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+// ========== Variables ========== //
+// VARIABLES: defaults
+var varWorkloadNameFormat = '${solutionPrefix}-{0}' //${uniqueString(resourceGroup().id, solutionPrefix)}
+var avmDeploymentNameFormat = 'ckm-deploy-avm-{0}'
+var localModuleDeploymentNameFormat = 'ckm-deploy-module-{0}'
+
+// VARIABLES: calculated resource names
+
+var varAiFoundryAiHubResourceName = empty(aiFoundryAiHubResourceName)
+  ? format(varWorkloadNameFormat, 'aifd-aihb')
+  : aiFoundryAiHubResourceName
+var varAiFoundryAiProjectResourceName = empty(aiFoundryAiProjectResourceName)
+  ? format(varWorkloadNameFormat, 'aifd-aipj')
+  : aiFoundryAiProjectResourceName
+var varAiFoundryAiServicesContentUnderstandingResourceName = empty(aiFoundryAiServicesContentUnderstandingResourceName)
+  ? format(varWorkloadNameFormat, 'aifd-aisr-cu')
+  : aiFoundryAiServicesContentUnderstandingResourceName
+var varAiFoundryAiServiceResourceName = empty(aiFoundryAiServicesResourceName)
+  ? format(varWorkloadNameFormat, 'aifd-aisr')
+  : aiFoundryAiServicesResourceName
+var varAiFoundryApplicationInsightsResourceName = empty(aiFoundryApplicationInsightsResourceName)
+  ? format(varWorkloadNameFormat, 'aifd-appi')
+  : aiFoundryApplicationInsightsResourceName
+var varAiFoundryContainerRegistryResourceName = empty(aiFoundryContainerRegistryResourceName)
+  ? replace(format(varWorkloadNameFormat, 'aifd-creg'), '-', '') //NOTE: ACR name should not contain hyphens
```

Comment:
Why not directly code this into the default of the parameter?

Reply:
So I guess the only working alternative would be something like
```bicep
@description('Optional. The name of the AI Foundry Application Insights resource. It will override the default given name.')
@maxLength(90)
param aiFoundryApplicationInsightsResourceName string = format('${solutionPrefix}-{0}', 'aifd-appi') // OR
param aiFoundryApplicationInsightsResourceName string = '${solutionPrefix}-aifd-appi'
```
requiring you to repeat the `varWorkloadNameFormat`?

To be honest - I'd probably recommend doing that as it will also save you a lot of space. Let me know what you think

Reply:
Addressed. I've followed the recommendation and added the default value directly. Also, I've updated the parameters with UDT per resource